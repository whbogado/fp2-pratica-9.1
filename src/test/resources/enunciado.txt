
Crie um projeto Maven no Netbeans clonando o repositório https://github.com/whbogado/fp2-template-basico.git.
Modifique o nome do grupo para utfpr.ct.dainf.if62c.pratica, o nome do artefato para Pratica-7.2 e o nome do projeto para FP2 :: Prática :: 7.2.
Dentro do pacote padrão crie a classe Pratica72 que implementará o seu programa principal.
Dentro pacote utfpr.ct.dainf.if62c.pratica crie a classe ContadorPalavras. Esta classe deve ter um construtor público que receba o caminho de um arquivo de texto cujas palavras serão contadas.
Na classe ContadorPalavras declare a propriedade privada reader do tipo BufferedReader. Esta propriedade deve ser inicializada no construtor usando o arquivo recebido como argumento.
Na classe ContadorPalavras implemente o método público getPalavras(). O objetivo é contar o número de ocorrências de cada palavra neste arquivo. Pontuação e espaços em branco devem ser ignorados. Este método deve retornar um mapa cuja chave seja a palavra e cujo valor seja o número de ocorrências (inteiro). Usando o reader, o método deve ler o arquivo, contar as palavras e preencher o mapa. Assegure que o arquivo seja fechado após a leitura.
No programa principal, solicite ao usuário que digite o nome (caminho completo) de um arquivo de texto. Instancie um ContadorPalavras com este arquivo. Use o método getPalavras() para obter as palavras encontradas.
Grave em outro arquivo texto a lista de palavras e o número de ocorrências em formato CSV, ou seja, cada linha do arquivo gerado deve conter palavra,número de ocorrências. Antes de gravar o arquivo, ordene as palavras pelo número de ocorrências, em ordem descendente. O arquivo de saída deve ser gerado na mesma pasta e ter o nome original mais o sufixo .out, por exemplo, se o arquivo de entrada era /tmp/livro.txt, o arquivo de saída deve ser /tmp/livro.txt.out.

