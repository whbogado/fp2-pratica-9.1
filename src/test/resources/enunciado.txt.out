o,26
de,15
arquivo,13
deve,7
a,6
e,6
do,6
um,6
nome,5
ser,5
n,5
palavras,5
ncias,5
ocorr,5
classe,5
mero,5
seja,4
m,4
todo,4
em,4
ContadorPalavras,4
que,4
para,4
as,3
palavra,3
texto,3
,2
construtor,2
programa,2
tmp,2
rio,2
if62c,2
ter,2
2,2
7,2
utfpr,2
sa,2
O,2
out,2
gerado,2
Dentro,2
propriedade,2
p,2
ct,2
txt,2
cada,2
pratica,2
da,2
com,2
reader,2
Esta,2
blico,2
projeto,2
livro,2
mapa,2
no,2
principal,2
Na,2
contar,2
crie,2
dainf,2
pacote,2
getPalavras,2
caminho,2
Maven,1
tipo,1
retornar,1
branco,1
completo,1
cuja,1
padr,1
pelo,1
implementar,1
reposit,1
ler,1
ordem,1
cujo,1
https,1
Pratica72,1
como,1
github,1
Pr,1
obter,1
exemplo,1
ao,1
ap,1
neste,1
espa,1
usando,1
template,1
digite,1
objetivo,1
artefato,1
por,1
se,1
Assegure,1
era,1
outro,1
clonando,1
conter,1
seu,1
inicializada,1
Use,1
valor,1
linha,1
fechado,1
Antes,1
s,1
entrada,1
whbogado,1
BufferedReader,1
mesma,1
sufixo,1
declare,1
ordene,1
grupo,1
solicite,1
mais,1
Netbeans,1
receba,1
Modifique,1
usu,1
Pratica,1
Este,1
lista,1
Pontua,1
tica,1
gravar,1
fp2,1
FP2,1
recebido,1
este,1
na,1
Usando,1
encontradas,1
cujas,1
basico,1
formato,1
Crie,1
ignorados,1
devem,1
privada,1
implemente,1
git,1
pasta,1
descendente,1
No,1
Instancie,1
original,1
os,1
inteiro,1
ou,1
CSV,1
chave,1
contadas,1
leitura,1
Grave,1
preencher,1
argumento,1
